pkg/docker/container_test.go

```diff
@@ -105,3 +105,69 @@ func TestContainerImageNameTagFromImage(t *testing.T) {
 		})
 	}
 }
+
+func TestContainerImageNameFromImage(t *testing.T) {
+	tt := []struct {
+		image         string
+		expected      string
+		expectedError string
+	}{
+		{image: "", expected: "", expectedError: "failed to parse image name"},
+		{image: "onlyImage", expected: "onlyImage"},
+		{image: "onlyimage", expected: "onlyimage"},
+		{image: "onlyimage:withTag", expected: "onlyimage"},
+		{image: "onlyimage@sha256:152f65865ae43b143b1e42dacdb5e9c473dd70b3adc5b79af7cf585cc8605205", expected: "onlyimage@sha256"},
+		{image: "path/to/image", expected: "path/to/image"},
+		{image: "my.registry.com/onlyimage", expected: "onlyimage"},
+		{image: "my.registry.com:50000/onlyimage", expected: "onlyimage"},
+		{image: "my.registry.com:50000/onlyimage:withTag", expected: "onlyimage"},
+		{image: "my.registry.com:50000/onlyimage@sha256:152f65865ae43b143b1e42dacdb5e9c473dd70b3adc5b79af7cf585cc8605205", expected: "onlyimage@sha256"},
```

Comment:
why do we expect `onlyimage@sha256` ?

Reply:
For the gitops use case this is fine as the default template adds exactly the missing colon. But in general case this looks weird and wrong. I will see if I can come up with something better.